Feature/연습용

kyeongminkang opened the pull request:
linq, boards 변수명, GetComments 메소드명 변경
GetComments() 메소드와 linq식,
BoardModelView 이름변경, CommentModelView 생성
쓸모없던 코드들 삭제(Bind["~"], creat에서 페이징 등)
모델 유효성검사 수정, 변수명 수정
CommentCounter (viewmodel) 없애고 BoardViewModel로 합침.
searchString -> searchstring으로 변경
Comment, List<Comment> 둘 다 있었는데 List만 남김.
[FromRoute],[FromQuery] 
PaginatedList 수정

Commit:
fix member variable naming format

kyeongminkang committed:
브렌치 되는 지 볼 겸.
linq, boards 변수명, GetComments 메소드명 변경

kyeongminkang committed:
GetComments() 메소드와 linq식,
BoardModelView 이름변경, CommentModelView 생성

kyeongminkang committed:
코드를 수정하다가 갑자기 오류가 터져서 먼 산으로 다녀오게 됨...
쓸모없던 코드들 삭제(Bind["~"], creat에서 페이징 등)

kyeongminkang committed:
모델 유효성검사 수정, 변수명 수정

Route , Nullable 모르겠음 할 것!!!!

kyeongminkang committed:
CommentCounter (viewmodel) 없애고 BoardViewModel로 합침.
searchString -> searchstring으로 변경
Comment, List<Comment> 둘 다 있었는데 List만 남김.

kyeongminkang committed:
[FromRoute],[FromQuery] 하면서 매개변수와 라우팅 일치시켰음.
그런데 컨트롤러에서 메소드 1개만 모두 Query로 받는데 Route로 받는 게 있어서 확인 중.

kyeongminkang committed:
PaginatedList 수정
Route 이상했던것(어느 부분은 Query고 어느부분은 Route) 수정

kyeongminkang committed:
윽..주석 안 지움..

kyeongminkang committed:
Merge branch 'feature/연습용'

# Conflicts:
#	MvcBoardApp/MvcBoardApp/Controllers/BoardsController.cs
#	MvcBoardApp/MvcBoardApp/Controllers/CommentsController.cs

kyeongminkang committed:
수정? 되는건가..

devtedlee commented on MvcBoardApp/MvcBoardApp/Controllers/BoardsController.cs:
```diff
@@ -15,137 +15,129 @@ namespace MvcBoardApp.Controllers
         [HttpGet]
-
-        public async Task<IActionResult> Index(string searchString, string sortOrder, string currentFilter, int? pageNumber)
+        public async Task<IActionResult> Index(string searchstring, string sortOrder, string currentFilter, int? pageNumber)
```
[FromQuery] 추가

devtedlee replied:
추가 안됐는데요?

devtedlee commented on MvcBoardApp/MvcBoardApp/Controllers/BoardsController.cs:
```diff
@@ -15,137 +15,129 @@ namespace MvcBoardApp.Controllers
             ViewData["CuurentSort"] = sortOrder;
 
-            if (searchString != null)
+            if (searchstring != null)
```
searchString 이 맞는거같은데 왜 바꿨어요?

kyeongminkang replied:
if (!String.IsNullOrEmpty(searchString)) 여기에서 수정하라는 뜻인데 착각했습니다. 수정했습니다.

devtedlee replied:
수정한건 Resolve conversation 버튼을 눌러주시면 됩니다.

devtedlee commented on MvcBoardApp/MvcBoardApp/Controllers/BoardsController.cs:
```diff
@@ -15,137 +15,129 @@ namespace MvcBoardApp.Controllers
 
-            var board = from m in _context.Board
-                        select m;
+            var boards = mDbContext.Board.AsQueryable();
```
보통 DbSet<Board> property의 이름은 Boards로 복수형을 해주는게 더 명확해요. 실제로 여러개 저장되 있는곳에서 가져오는 거잖아요? Comment도 마찬가지고 어떤 단어라도 복수형 쓸 수 있는 단어면 복수형 쓰는게 더 좋습니다.

devtedlee replied:
읽기 전용 엔티티 Queryable이면 여기에 .AsNoTracking() 붙여주면 됩니다. 굳이 제일 마지막에 붙일 이유가 전혀 없고 오히려 더 혼란스럽죠.

kyeongminkang replied:
제가 저번 커멘트에서 이해를 제대로 하고 넘어간 줄 알았는데 잘못 이해한 부분이 있는 것 같아 이 곳에 질문합니다.  
"boards라고 써져있던 이유는 board entity가 복수개이기때문에 boards라고 복수형으로 네이밍 된거에요. 다른곳에서 board라고 쓴거는 board entity가 단수이기 때문에 board 라고 네이밍 된거구요." 
이것은 변수를 말씀하신것이며, Db에서 복수의 entity라는 것이 예를 들어 id를 가져오는데 id가 조건(where) 으로 인해 여러 id를 가져오거나, 
id 1개가 아닌 id, subject, content 이런 식으로 여러개를 가져오는 것이라 생각한 것이 맞을까요?

kyeongminkang replied:
또한, 이번 커멘트에서는 말씀하신 Dbset의 프로퍼티가 결국 테이블 명(위에서 고치라고 하신 Models의 Entity에 들어가는 클래스들)을
그 단어가 복수형이 되지 않는 경우를 제외하고는 거의 무조건 복수형으로 네이밍 하는 것으로 이해하는 것이 맞을까요?
그렇다면 ViewModel에서 모델 객체를 통해 데이터를 전달할 때? 모델 객체를 클래스로 받아올 때? (표현을 어떻게 해야할 지 모르겠습니다..)
일반 클래스는 단수, 리스트로 된 것은 복수로 해야한다고 들었는데
예를 들어 public Boards board{get; set;} public List"<"Comments">"comments {get; set;} 이렇게 해야하는 건가요?

kyeongminkang replied:
AsQueryable()대신 AsNoTracking()을 사용하라는 뜻은 AsNoTracking()은 캐시에 남지 않아 메모리 사용을 줄여주기 때문에 읽기 전용으로 사용하기에 적합하고,
AsQueryable의 경우는 IQueryable로 반환하여 대개 뒤에 조건(where, select와 같은)과 함께 쓰기 때문인가요?

kyeongminkang replied:
AsQueryable이 굳이 조건이랑 같이 쓰고말고의 문제가 아님! 대개, Get사용 시 사용하면 좋은데, 다른 쿼리들은 값을 가져오고, 그대로 연결되어 있기 때문에 수정 및 삭제를 가능하게 해줌. 하지만 AsNoTracking의 경우, 읽기전용이기 때문에 값을 한 번 딱 가져오고 끝임.  값 변경을 못한다는 뜻! 그렇기 때문에 실수도 줄여주고, 메모리도 아껴줌.

devtedlee commented on MvcBoardApp/MvcBoardApp/Controllers/BoardsController.cs:
```diff
@@ -15,137 +15,129 @@ namespace MvcBoardApp.Controllers
-        [Route("Details/{id}")]
-        public async Task<IActionResult> Details(int? id, int? pageNumber)
+        [Route("Details")]
+        public async Task<IActionResult> Details([FromQuery]int? ID, [FromQuery]int pageNumber)
```
page Number가 필수라면 라우팅에 {pageNumber}로 추가시키는게 더 이해하기 좋습니다. 굳이 nullable이 될 수 있는 매개변수 뒤에 올 이유가 없음.

그리고 무엇보다 ID자체를 라우팅 매개변수로 안넣은게 이상하네요. 디테일 페이지에 ID가 빠지면 말이 안돼잖아요?

devtedlee replied:
이거 ID를 라우팅으로 넣기로 하지않았나

devtedlee commented on MvcBoardApp/MvcBoardApp/Controllers/BoardsController.cs:
```diff
@@ -15,137 +15,129 @@ namespace MvcBoardApp.Controllers
                 Board = board,
-                Comments = GetComment(id),
-                PageIndex = (int)pageNumber
+                Comments = mDbContext.Comment.Where(m => m.BoardID == ID).ToList(),
```
이렇게 데이터베이스에서 가져오는 값을 바로 연결시키면 에러가 나도 어디에서 에러가 나는지 디버거로 추적하기가 어렵습니다. 

`List<Comment> comments = await DbContext.Comments.Where(m => m.BoardID == ID).ToListAsync();` 이렇게 빼주세요.

devtedlee replied:
async await 잘 모르면 참고 https://docs.microsoft.com/ko-kr/dotnet/csharp/programming-guide/concepts/async/

devtedlee replied:
여기 컨트롤러 메소드 자체 리턴 타입을 async Task<IActionResult> 로 변경 해줘야합니다.

kyeongminkang replied:
비동기란 한 스레드로 다른 작업을 차단하지 않고 Async, await 키워드를 이용하여 실행을 계속 시키는 것이며
메소드 자체 리턴타입을 async Task로 하라고 하신 것은 Task<T> 즉 값을 반환하는 비동기 메소드라는 뜻이라 이해하는 것이 맞을 까요?
제가 자체 리턴타입이란 말의 뜻을 이해하지 못하여 RedirectToAction을 사용하였는데, 
이는 두 가지 가능한 반환 형식이 포함된 것이므로 현재 async Task<IActionResult> Details 메소드에서는 그대로 모델뷰를 리턴하는 것이 맞는 지 궁금합니다.

devtedlee replied:
컨트롤러 자체 메소드 == BoardsController.Details 메소드

devtedlee commented on MvcBoardApp/MvcBoardApp/Controllers/BoardsController.cs:
```diff
@@ -15,137 +15,129 @@ namespace MvcBoardApp.Controllers
         [HttpGet]
         [Route("Create")]
-        public async Task<IActionResult> Create(int? pageNumber)
+        public IActionResult Create([FromQuery]int pageNumber)
```
pageNumber 굳이 필요없을거같은데.. 따로 의도가 있나요?

kyeongminkang replied:
그냥 글만 적을 때는 무조건 1페이지로 넘어가서 상관없지만, Create.cshtml에서 List로 돌아가기 위해(Index) pageNumber값을 같이 보내주고 있었습니다.

devtedlee commented on MvcBoardApp/MvcBoardApp/Controllers/BoardsController.cs:
```diff
@@ -15,137 +15,129 @@ namespace MvcBoardApp.Controllers
         [ValidateAntiForgeryToken]
         [Route("Create")]
-        public async Task<IActionResult> Create([Bind("ID,UserName,Subject,Content,WriteDate,CommentCount")] Board board, int? pageNumber)
+        public async Task<IActionResult> Create(Board board, [FromQuery]int pageNumber)
```
Board board 에 [FromForm] 추가

devtedlee commented on MvcBoardApp/MvcBoardApp/Controllers/BoardsController.cs:
```diff
@@ -15,137 +15,129 @@ namespace MvcBoardApp.Controllers
             {
-                _context.Add(board);
-                await _context.SaveChangesAsync();
+                mDbContext.Add(board);
```
이거 잘못됐네요. 보통 Form에서 입력한거 DB에 입력하려면 그 입력값들을 검증하는 과정이 추가되야 합니다. 그러려면 DB 엔티티와 동일한 구조의 모델을 사용하지 않아요.

즉, 이거 CreateBoardViewModel 같은 클래스 따로 추가해서 가져온 후 입력 값들 검증하고, Board 엔티티로 검증된 값들 옮겨준다음 그걸 DB에 추가해줍니다.

이해안돼면 질문 더 하시길

kyeongminkang replied:
현재 Board와 Comment 모두 CreateBoardViewModel, CreateCommentViewModel로 ViewModel을 만들었습니다.  필요한 DB 엔티티만 가져온 후, ViewModel에서 [Required] 특성을 주고, cshtml에서 a asp-validation 태그를 사용한 후, Controller에서 if (ModelState.IsValid) 메소드를 사용하였습니다. 이렇게 하는 것이 입력 값을 검증하는 것이 맞는 지 궁금합니다.

kyeongminkang replied:
또한, Board 엔티티로 검증된 값들을 옮겨준 다음, DB에 추가하라고 하셨는데 예를 들면 Board를 new로 새 객체를 생성한 후, mDbContext.Add를 하라는 말씀이 맞으신지요?

kyeongminkang replied:
또한, Board 엔티티로 검증된 값들을 옮겨준 다음, DB에 추가하라고 하셨는데 예를 들면 Board를 new로 새 객체를 생성한 후, ViewModel에 있는 값을 넣어주고, mDbcontext.Add를 하라는 말씀이 맞으신지요?

kyeongminkang replied:
아 그리고 혹시 주말에 알람이 울렸다면 죄송합니다...

kyeongminkang replied:
추가 질문드립니다. 제가 이해한 것으로는 DB에 함부로 직접 접근하지 못하게 하기 위해 Create와 Edit의 경우 ViewModel을 생성하는 것 같습니다. Delete같은 경우, Form과 상관없이 삭제되는 것이기 때문에 ViewModel이 필요없다고 이해하는 것이 맞을까요?

devtedlee replied:
네, 클라이언트에서 입력할 고유한 값이 없으니까요

devtedlee commented on MvcBoardApp/MvcBoardApp/Controllers/BoardsController.cs:
```diff
@@ -15,137 +15,129 @@ namespace MvcBoardApp.Controllers
-        [Route("Edit/{id}")]
-        public async Task<IActionResult> Edit(int? id, [Bind("ID,UserName,Subject,Content,WriteDate,CommentCount")] Board board, int pageNumber)
+        [Route("Edit")]
+        public async Task<IActionResult> Edit([FromQuery]int? ID, Board board, [FromQuery]int pageNumber)
```
이것도 마찬가지. ID는 필수

devtedlee commented on MvcBoardApp/MvcBoardApp/Controllers/BoardsController.cs:
```diff
@@ -154,12 +146,12 @@ public async Task<IActionResult> Edit(int? id, [Bind("ID,UserName,Subject,Conten
                 {
-                    _context.Update(board);
-                    await _context.SaveChangesAsync();
+                    mDbContext.Update(board);
```
보통입력된 Board 뷰모델이 유효하면,  entity Db에서 가져와서 값을 수정하고 SaveChanges()만 해주면 됩니다.

EditBoardViewModel 클래스 필요

kyeongminkang replied:
위에  await과 비슷한 질문인 것 같습니다. 보내주신 사이트와 다른 문서들을 읽어도 여전히 이해가 부족해서 또 질문 드립니다...
Savechanges만 사용해도 된다고 하신 것은 다른 작업을 대기할 필요가 없기 때문일까요? 
아니면 await 요청중에 다른 작업으로 인해 값이 바뀔 수 있기 때문에 Edit은 동기식으로 하는것인 지 
아니면 제가 이해하는 방향이 아예 틀린 것인지 궁금합니다.

devtedlee replied:
제가 말한 방식대로 하면, 원래는 `mDbContext.Edit()` 이런식의 값 수정 전용 메소드가 SaveChanges() 메소드 위에 있었는데 그 위에 있던거 없이 SaveChanges만 사용해도 된다는 거였습니다.

devtedlee commented on MvcBoardApp/MvcBoardApp/Controllers/BoardsController.cs:
```diff
@@ -168,50 +160,52 @@ public async Task<IActionResult> Edit(int? id, [Bind("ID,UserName,Subject,Conten
             }
 
-            var board = await _context.Board.FirstOrDefaultAsync(m => m.ID == id);
+            var board = await mDbContext.Board.FirstOrDefaultAsync(m => m.ID == ID);
```
var 보다 Board 라고 명시하는걸 더 좋아합니다. var는 같은 줄에 new Board 처럼 `=`의 우측을 봤을 때 타입을 알 수 있는 경우로 제한해서 사용해요. (회사 코딩스탠다드)

devtedlee commented on MvcBoardApp/MvcBoardApp/Controllers/BoardsController.cs:
```diff
@@ -168,50 +160,52 @@ public async Task<IActionResult> Edit(int? id, [Bind("ID,UserName,Subject,Conten
         [ValidateAntiForgeryToken]
         [Route("Delete")]
-        public async Task<IActionResult> DeleteConfirmed(int? id, int? pageNumber)
+        public async Task<IActionResult> DeleteConfirmed(int? ID, [FromQuery]int pageNumber)
```
굳이 Delete [HttpGet] 페이지가 없는데 이름을 DeleteConfirmed라고 바꿔줄 필요는 없습니다. 그럼 ActionName("Delete")도 없애도 되구요.

devtedlee commented on MvcBoardApp/MvcBoardApp/Controllers/BoardsController.cs:
```diff
@@ -168,50 +160,52 @@ public async Task<IActionResult> Edit(int? id, [Bind("ID,UserName,Subject,Conten
         [ValidateAntiForgeryToken]
         [Route("Delete")]
-        public async Task<IActionResult> DeleteConfirmed(int? id, int? pageNumber)
+        public async Task<IActionResult> DeleteConfirmed(int? ID, [FromQuery]int pageNumber)
```
ID는 필수, 그리고 FromRoute나 FromQuery 둘중 하나도 없음

kyeongminkang replied:
전 글에 "url형식이 쿼리인것과 Route를 몰라서 이해를 하지 못 했습니다. 제가 공부한 바로는 url형식이 쿼리인 것은 https://localhost/Board?ID=1&pageNumber=3 이런 형태이고, Route는 https://localhost/Board/ID/pageNumber 파라미터가 들어가면 https://localhos/Board/1/3 이런 형태인 것 같습니다. 그렇다면 이 pageNumber는 쿼리 형태인 것이며, 어느 곳에서든 pageNumber의 값이 필요하므로 필수적인 쿼리라고 이해하는게 맞을까요? 우선 제가 이해한 방식으로 해서 라우팅 매개변수와 메소드 매개변수를 일치 시켰습니다." 이렇게 여쭈어봤었습니다. 
그래서 방금처럼 빼놓은 건 제외하고 [FromQuery]를 달았었는데, 필수라고 하시는 것이Route["Create/{pageNumber}"] 이러한 형태를 말씀하시는 것이 맞나요?  Url쿼리와 Route를 왜 다르게 써야하는 지 이해를 하지 못하고 있습니다. 일단 Create를 제외하고는 Id, pageNumber가 무조건 있어야할 것 같아 모든 Route[/{ID}/{pageNumber}]에 붙여놨습니다.

devtedlee commented on MvcBoardApp/MvcBoardApp/Controllers/BoardsController.cs:
```diff
@@ -168,50 +160,52 @@ public async Task<IActionResult> Edit(int? id, [Bind("ID,UserName,Subject,Conten
         }
 
-        private bool BoardExists(int id)
+        private bool BoardExists(int ID)
```
코딩 스탠다드에서 private 메소드는 어떤 네이밍 규칙을 써야하나 다시 찾아보시길

devtedlee commented on MvcBoardApp/MvcBoardApp/Controllers/CommentsController.cs:
```diff
@@ -14,91 +14,94 @@ namespace MvcBoardApp.Controllers
+                Board board = mDbContext.Board.FirstOrDefault(m => m.ID == comment.BoardID);
 
-                var commentCounter = new CommentCounter
+                var boardViewModel = new BoardViewModel()
```
BoardViewModel이 왜 필요해요?

그리고 이미 Count 가져왔는데 또 GetCount(board)이건 왜 하는거에요?

kyeongminkang replied:
수정하였습니다.

devtedlee commented on MvcBoardApp/MvcBoardApp/Controllers/CommentsController.cs:
```diff
@@ -14,91 +14,94 @@ namespace MvcBoardApp.Controllers
         }
 
-        public int Counter(int? id)
+        public int Counter(int? ID)
```
이거 여기 클래스 외부에서 쓰는 메소드인가요?

devtedlee commented on MvcBoardApp/MvcBoardApp/Controllers/CommentsController.cs:
```diff
@@ -14,91 +14,94 @@ namespace MvcBoardApp.Controllers
         [HttpGet]
         [Route("Edit")]
-        public async Task<IActionResult> Edit(int? id, int? pageNumber)
+        public async Task<IActionResult> Edit([FromQuery]int? ID, [FromQuery]int pageNumber)
```
ID 필수

devtedlee commented on MvcBoardApp/MvcBoardApp/Controllers/CommentsController.cs:
```diff
@@ -14,91 +14,94 @@ namespace MvcBoardApp.Controllers
             }
 
-            BoardComment boardComment = new BoardComment
+            CommentViewModel commentViewModel = new CommentViewModel
```
.... ViewModel에 Comment 엔티티 그대로 사용하면 안됩니다. 같은 prop들을 가지고 있어야 되요.

kyeongminkang replied:
prop가 프로퍼티 맞을까요?
ViewModel에 Comment 엔티티를 그대로 사용하면 안된다는 말씀이 
예를 들면 ViewModel에 Comment {get; set;}이 아닌 Comment 엔티티안에 프로퍼티인 UserName {get; set;}을 사용하라는 뜻인 지 궁금합니다.
현재 코멘트 주신 곳은 Create()였는데, Detail이나 Index 같은 경우는 Db에 직접 접근하지 않기 때문에 그대로 사용해도 되는 건가요?

devtedlee replied:
글루와 프로젝트 코드 보고 참고하세요.

devtedlee commented on MvcBoardApp/MvcBoardApp/Controllers/CommentsController.cs:
```diff
@@ -14,91 +14,94 @@ namespace MvcBoardApp.Controllers
         [ValidateAntiForgeryToken]
         [Route("Edit")]
-        public async Task<IActionResult> Edit(int? id, [Bind("ID,CommentUserName,CommentContent,BoardID")] Comment comment, int? pageNumber)
+        public async Task<IActionResult> Edit([FromQuery]int? ID, Comment comment, [FromQuery]int pageNumber)
```
CommentEditViewModel

devtedlee replied:
이거 안고쳤잖아요

devtedlee commented on MvcBoardApp/MvcBoardApp/Controllers/CommentsController.cs:
```diff
@@ -122,72 +125,72 @@ public async Task<IActionResult> Edit(int? id, [Bind("ID,CommentUserName,Comment
         [HttpGet]
         [Route("Delete")]
-        public async Task<IActionResult> Delete(int? id, int? pageNumber)
+        public async Task<IActionResult> Delete([FromQuery]int? ID, [FromQuery]int pageNumber)
```
ID 필수

kyeongminkang replied:
전 커멘트에 질문을 남겼었는데 제가 이해하고 있는게 맞다면 필수로 넣으려면 Route에 {}해서 넣어주는 것으로 알고 있는데 Id와(Create는 제외) pageNumber 둘 다 필수로 넣기 위해 모두 Route에 넣어주면서 [FromQuery]를 다 빼주었는데 방법이 맞는 지 궁금합니다.

devtedlee commented on MvcBoardApp/MvcBoardApp/Migrations/MvcBoardAppContextModelSnapshot.cs:
```diff
@@ -21,17 +21,19 @@ protected override void BuildModel(ModelBuilder modelBuilder)
                     b.Property<string>("Content")
-                        .IsRequired();
+                        .IsRequired()
+                        .HasMaxLength(100);
```
100자는 너무 짧은거같은데..

devtedlee reviewed (commented):
ViewModels 폴더는 Models폴더 하위에. Models에 바로 담겨있는 Entity들은 Models/Entity 폴더에 넣어주세요. namespace 바꾸는거 잊지 마시구요.

kyeongminkang committed:
models에 폴더 정리(Entity, ViewModels)
매개변수 searchString으로 다시 정정
잘못 사용한 자료형 명변경(String->string)
Dbset 프로퍼티 변경
AsNotracking 추가

kyeongminkang committed:
마이그레이션 오류
model에서 string 변수 변경

kyeongminkang committed:
라우팅 변경 (어떻게 될 지 모르겠음)
fromform 추가
Delete 메소드 명 변경
private 메소드 변경
CommentCount 변경 (기존 BoardViewModel에서 제외)

kyeongminkang committed:
var 키워드 변경
Route, url Query 처리 완료(코멘트에서는 Query로, 글은 그대로 Route처리함. 글 수정하고, 삭제했을 때 본래 페이지로 와야하니까)

나머지는 정리 다 됐는데 ViewModel을 하나하나 만들어줘야하고 ViewModel 유효성검사를 해야하는데 어떻게 해야할지 고민과 공부를 해야겠....ㅜㅜ

kyeongminkang committed:
Board Create에서 List로 돌아가는 게 오류나서 살펴보니
1페이지로 무조건 보내기로 해놓고 또 page를 넘겨줘서 안 됐던 것 같음.
ViewModel 작업을 하고 있는데
ViewModel을 각각 따로 만들어서 유효성 검사를 해야하는데 현재 하고 있는 게 맞는지 모르겠음....
또한 아직도 async await 비동기식 처리를 모르겠음...
금요일에 까먹고 시간 안 적은 것 같은데 퇴근시간에서 1시간 반 더 작업했고
주말에는 4시간 반정도!

kyeongminkang committed:
Back to List와 Back to Previous로 분리
ViewModel 추가 및 수정

kyeongminkang committed:
namespace 변경

kyeongminkang committed:
Merge branch 'master' into feature/연습용

# Conflicts:
#	MvcBoardApp/MvcBoardApp/Controllers/CommentsController.cs

devtedlee reviewed (approved):
이정도로 마무리 하죠. 이제 Complete 하시면 현재 브랜치의 변경사항이 master 브랜치로 머지될거에요.

devtedlee reviewed (commented):
리뷰완료
